### PR TITLE
Use a vergen that handles spew in timestamp output

### LINKED ISSUE #1531: Install Spin - Error
Hi
Just noticing an issue when using the docs installation procedure.

```bash
$ git clone https://github.com/fermyon/spin -b v1.2.1
$ cd spin
$ rustup target add wasm32-wasi
$ rustup target add wasm32-unknown-unknown
$ cargo install --locked --path .
$ spin --help

error: failed to run custom build command for `spin-cli v1.2.0 (/Users/tpmccallum/spin)`

Caused by:
  process didn't exit successfully: `/Users/tpmccallum/spin/target/release/build/spin-cli-1019c0ea55ff8bd4/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'failed to extract build information: the 'year' component could not be parsed

  Caused by:
      the 'year' component could not be parsed', build.rs:28:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `spin-cli v1.2.0 (/Users/tpmccallum/spin)`, intermediate artifacts can be found at `/Users/tpmccallum/spin/target`
```
Looks like `failed to extract build information` error.

### PR DESCRIPTION
**Pull request by itowlson**
Fixes #1531.

cc @tpmccallum - I think you are the only person who was able to repro this - if you have a moment to check the fix, please try:

```
cargo install spin-cli --git https://github.com/itowlson/spin --branch vergen-timestamp-update --locked
```

Thanks!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -87,7 +87,7 @@ e2e-testing = { path = "crates/e2e-testing" }
 
 [build-dependencies]
 cargo-target-dep = { git = "https://github.com/fermyon/cargo-target-dep", rev = "b7b1989fe0984c0f7c4966398304c6538e52fe49" }
-vergen = { version = "8", default-features = false, features = [
+vergen = { version = "^8.2.1", default-features = false, features = [
```
Comment (lann) [inline Cargo.toml:90]: FYI

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements

> Caret requirements are an alternative syntax for the default strategy, ^1.2.3 is exactly equivalent to 1.2.3.

**Review by lann — APPROVED** [review]
You can achieve this without changing `Cargo.toml` with e.g. `cargo update -p vergen --precise 8.2.1`, but I don't really know when one would prefer one over the other.

**Review by fibonacci1729 — APPROVED** [review]

**Comment by itowlson** [comment]
@lann Thank you - TIL!  I'm going to leave this as is if it works for @tpmccallum, but useful to know for the future!

**Comment by tpmccallum** [comment]
Voila - works great; thanks @itowlson 

Just to be sure I began from scratch and tried the normal installation - which failed again expectedly.

I then tried your new command, and it installed Spin without any issues.


spin

**Comment by itowlson** [comment]
Thanks @tpmccallum!  (And thanks Vergen folks for such a quick fix!)

**Pull request merged**